src/renderer/services/transaction/callDataDecoder/callDataDecoder.ts

```diff
@@ -0,0 +1,505 @@
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    return {};
+  }
+
+  supports(): TransactionType {
+    return TransactionType.CHILL;
+  }
+}
+
+class StakingRestakeCallDataArgsParser extends AbstractCallDataArgsParser {
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    return {
+      value: decoded.args[0].toString(),
+    };
+  }
+
+  supports(): TransactionType {
+    return TransactionType.RESTAKE;
+  }
+}
+
+class StakingRedeemCallDataArgsParser extends AbstractCallDataArgsParser {
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    return {};
+  }
+
+  supports(): TransactionType {
+    return TransactionType.REDEEM;
+  }
+}
+
+class StakingNominateCallDataArgsParser extends AbstractCallDataArgsParser {
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    return {
+      targets: (decoded.args[0] as any).map((a: Type) => a.toString()),
+    };
+  }
+
+  supports(): TransactionType {
+    return TransactionType.NOMINATE;
+  }
+}
+
+class StakingStakeMoreCallDataArgsParser extends AbstractCallDataArgsParser {
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    return {
+      maxAdditional: decoded.args[0].toString(),
+    };
+  }
+
+  supports(): TransactionType {
+    return TransactionType.STAKE_MORE;
+  }
+}
+
+class StakingChangeDestinationCallDataArgsParser extends AbstractCallDataArgsParser {
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    const args: Record<string, any> = {};
+    try {
+      args.payee = JSON.parse(decoded.args[0].toString());
+    } catch (e) {
+      console.warn(e);
+      args.payee = decoded.args[0].toString();
+    }
+
+    return args;
+  }
+
+  supports(): TransactionType {
+    return TransactionType.DESTINATION;
+  }
+}
+
+class BatchAllCallDataArgsParser extends AbstractCallDataArgsParser {
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    return {
+      calls: decoded.args[0].toHex(),
+    };
+  }
+
+  supports(): TransactionType {
+    return TransactionType.BATCH_ALL;
+  }
+}
+
+class MultisigAsMultiCallDataArgsParser extends AbstractCallDataArgsParser {
+  public parseDecodedCallArgs(
+    method: string,
+    section: string,
+    decoded: SubmittableExtrinsic<'promise'>,
+  ): Record<string, any> {
+    if (decoded.args.length == 5) {
```

Comment:
Better to avoid to use `magic numbers` and use constants. Here you can use `OLD_MULTISIG_ARGS_AMOUNT` from `constants` file